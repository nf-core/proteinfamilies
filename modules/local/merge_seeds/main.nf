process MERGE_SEEDS {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/7d/7d0fee0217685dc2501570e1dd12076f9466175d0a37335ca424390cffec5fa1/data' :
        'community.wave.seqera.io/library/python:3.13.1--d00663700fcc8bcf' }"

    input:
    tuple val(meta) , val(similarities)
    tuple val(meta2), path(seed_msa, stageAs: "seed_msa/*")

    output:
    tuple val(meta), path("${prefix}*"), emit: merged_seed_msa
    path "versions.yml"                , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    merge_seeds.py \\
        --list "${similarities}" \\
        --folder seed_msa \\
        --out_file ${prefix}.fas

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version 2>&1 | sed 's/Python //g')
    END_VERSIONS
    """

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.fas

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version 2>&1 | sed 's/Python //g')
    END_VERSIONS
    """
}
