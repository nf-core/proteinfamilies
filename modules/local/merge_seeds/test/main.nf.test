nextflow_process {

    name "Test Process MERGE_SEEDS"
    script "../main.nf"
    process "MERGE_SEEDS"

    test("proteinfamilies - aln") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'test_similarities' ],
                    "mgnifams_test_1,mgnifams_test_2",
                ])
                input[1] = Channel.of([
                    [ id: 'test_aln' ],
                    [
                        file(params.pipelines_testdata_base_path + 'test_data/modules/mgnifams_test_1.aln', checkIfExists: true),
                        file(params.pipelines_testdata_base_path + 'test_data/modules/mgnifams_test_2.aln', checkIfExists: true)
                    ]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.merged_seed_msa,
                    process.out.versions.collect { path(it).yaml }
                    ).match()}
            )
        }
    }

    test("proteinfamilies - aln - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'test_similarities' ],
                    "mgnifams_test_1,mgnifams_test_2",
                ])
                input[1] = Channel.of([
                    [ id: 'test_aln' ],
                    [
                        file(params.pipelines_testdata_base_path + 'test_data/modules/mgnifams_test_1.aln', checkIfExists: true),
                        file(params.pipelines_testdata_base_path + 'test_data/modules/mgnifams_test_2.aln', checkIfExists: true)
                    ]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.merged_seed_msa,
                    process.out.versions.collect { path(it).yaml }
                    ).match()}
            )
        }

    }
}
