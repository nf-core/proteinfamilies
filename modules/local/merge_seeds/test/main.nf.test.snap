{
    "proteinfamilies - aln": {
        "content": [
            [
                [
                    {
                        "id": "test_similarities"
                    },
                    "null.fas:md5,19ae8f20eee8455d7f0f8e0a0dcfba93"
                ]
            ],
            [
                {
                    "MERGE_SEEDS": {
                        "python": "3.13.1"
                    }
                }
            ]
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.8"
        },
        "timestamp": "2025-10-09T10:44:51.541243546"
    },
    "proteinfamilies - aln - stub": {
        "content": [
            [
                [
                    {
                        "id": "test_similarities"
                    },
                    "null.fas:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                {
                    "MERGE_SEEDS": {
                        "python": "3.13.1"
                    }
                }
            ]
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.8"
        },
        "timestamp": "2025-10-09T10:44:58.843814558"
    }
}